projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -599,13 +599,11 @@ export class IgxDateRangePickerComponent extends DisplayDensityBase
 
     /** @hidden */
     public ngAfterViewInit(): void {
-        if (this.mode === InteractionMode.DropDown) {
-            this.attachOnKeydown();
-        }
         this.subscribeToDateEditorEvents();
         this.configPositionStrategy();
         this.configOverlaySettings();
         this.cacheFocusedInput();
+        this.attachOnKeydown();
```

Comment:
Just make this a `HostListener` at this point, since it's no longer under a condition. This will also let you easily properly test the open portion (which currently triggers on the (not visible) calendar.. to open it)